### ISSUE TITLE
ssh keys to github...

### ISSUE THREAD

**Issue by yina** [Furaha/ugandasms#40]
so close! stuck at trying to have the right permissions to get the repository from github. copied my id_rsa and id_rsa.pub files to the bootstrap directory and see that they are copied to the deploy user. However error with:

```
==> default: Cloning into '/home/deploy/app'...
==> default: Warning: Permanently added the RSA host key for IP address '192.30.252.129' to the list of known hosts.
==> default: Permission denied (publickey).
==> default: fatal: Could not read from remote repository.
==> default: 
==> default: Please make sure you have the correct access rights
==> default: and the repository exists.
```

**Comment by yina** [comment]
probably a NAT versus bridged network issue. looking in it.

**Comment by yina** [comment]
nope that wasn't it.

**Comment by arafatm** [comment]
Didn't think it would be. Are you using the deploy key that's set up for
the repo? Did I share that with you? Althought if you can 'git pull' then
you should already have the right keys

**Comment by arafatm** [comment]
Do you want to do a video call this evening? Unfortunately probably won't
be home before 7pm PST.

**Comment by yina** [comment]
I don't think I have the right keys. I can use my yina keys and get to the
repo and clone using the vagrant user but if I sudo -u deploy even with
copied keys from the vagrant user it doesn't work.

**Comment by yina** [comment]
could we gchat. i'll be on for about 30 minutes. till 10:30 EST.

**Comment by arafatm** [comment]
I'm in a meeting. tomorrow afternoon?

**Comment by yina** [comment]
available between 3 and 4 my time. i'm sure it's something totally dumb.

**Comment by yina** [comment]
oh i think it is dumb. i need the deploy user keys i think.

**Comment by yina** [comment]
some additional sleuthing. feels closer.

```
> vagrant ssh
> ssh -T git@github.com # this works
> sudo ssh -T git@github.com # this fails
> sudo SSH_AUTH_SOCK=$SSH_AUTH_SOCK ssh -T git@github.com # this works
> sudo -u deploy ssh -T git@github.com # this fails
> sudo -u deploy SSH_AUTH_SOCK=$SSH_AUTH_SOCK ssh -T git@github.com # this fails
```

**Comment by yina** [comment]
this looks promising: http://serverfault.com/questions/107187/ssh-agent-forwarding-and-sudo-to-another-user

**Comment by arafatm** [comment]
I sent a calendar invite. Let me know if that works.

**Comment by yina** [comment]
dang it... no luck. same behavior as above.

**Comment by arafatm** [comment]
hmm. I'll take another look.

**Comment by arafatm** [comment]
Did you git pull latest?

**Comment by yina** [comment]
yes… did so. i did the same procedure in the above comments and had the
same results. Yin
